src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipeline.scala

```diff
@@ -108,11 +108,17 @@ object ExtractionPipeline {
       linkingEntities = replicates
     )
 
+    val fcExperiments = extractLinkedEntities(
+      entityToExtract = EncodeEntity.FunctionalCharacterizationExperiment,
+      matchingField = "experiment",
+      linkingEntities = replicates
```

Comment:
This should have the correct behavior as-is, but it's sub-optimal. There will be batches of IDs that contain no `/experiment/`s, and other batches that contain no `/functional-characterization-experiment/`s. Those queries are guaranteed to return 0 results when we run them against the wrong `type=` value, so it'd be good for us (in runtime) and ENCODE (in server stability) if we avoided making them.

`SCollection` has a few `partition` methods that I think would be useful here: https://github.com/spotify/scio/blob/master/scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala#L262-L286. I think to reuse the plumbing, the flow will need to be:
1. `flatMap` the `replicates` stream so we end up with an `SCollection` of `JsonObject("experiment" -> "/<type>/<accession>")`
2. `partition` that stream to separate `type=experiment` from `type=functional-characterization-experiment`
3. Feed each partition into `extractLinkedEntities` calls for the corresponding type

Reply:
Created a new method to determine whether something is a functional-characterization-experiment or not. The method uses flatMap to flatten the replicate stream and turn it into a string and then evaluates the contents of the string. This method is working properly and correctly partitioning the replicates stream.